### PR TITLE
More vue converted to TypeScript

### PR DESCRIPTION
**Pull request by johnsimons**
I am not 100% happy with the generic `DataView` implementation.
I may raise an issue in vue to determine if there is a nicer way of doing that.

Apart from that, all the other conversions are straightforward.

### CODE REVIEW

File: src/ServicePulse.Host/vue/src/components/DataView.vue [hunk 1]
```diff
@@ -1,33 +1,26 @@
...
-    type: Boolean,
-    default: false,
-  },
-});
 
-const refreshTimer = ref();
-const items = ref([]);
+const props = withDefaults(
+  defineProps<{
+    itemsType: T[];
```
Comment (mikeminutillo) [inline src/ServicePulse.Host/vue/src/components/DataView.vue:11]: Would it make sense to turn this into a `v-model`? That way the caller would at least have access to the items if they needed them for some reason and the prop would not be completely irrelevant? Perhaps that would mean it is no longer necessary to pass the items to the slot. If we include total-count in the v-model then we would not need to pass anything in to the footer either.

File: src/ServicePulse.Host/vue/src/components/DataView.vue [hunk 2]
```diff
@@ -1,33 +1,26 @@
...
+    itemsPerPageOptions?: number[];
+    itemsPerPage?: number;
+    autoRefreshSeconds: number;
+    showPagination?: boolean;
+    showItemsPerPage?: boolean;
+  }>(),
+  { itemsPerPageOptions: () => [20, 35, 50, 75], itemsPerPage: 50, showPagination: true, showItemsPerPage: false }
+);
+
+const refreshTimer = ref<number>();
```
Comment (mikeminutillo) [inline src/ServicePulse.Host/vue/src/components/DataView.vue:22]: Does this default to 0? Is it OK to call `clearInterval` on a 0?

Comment (mikeminutillo) [reply src/ServicePulse.Host/vue/src/components/DataView.vue:22]: Also, I wrote this as a `ref` but I'm wondering if it should be a `let`. We don't bind this to anything.

File: src/ServicePulse.Host/vue/src/components/ExclamationMark.vue [hunk 3]
```diff
@@ -1,11 +1,16 @@
-﻿<script setup>
-const props = defineProps({
-  type: String,
-});
+﻿<script setup lang="ts">
+import { LicenseWarningLevel } from "@/composables/LicenseStatus";
+
+const props = withDefaults(
+  defineProps<{
+    type?: LicenseWarningLevel;
```
Comment (mikeminutillo) [inline src/ServicePulse.Host/vue/src/components/ExclamationMark.vue:6]: We definitely use this control in other contexts than only licenses. For instance, [we show it when we cannot connect to ServiceControl](https://github.com/Particular/ServicePulse/pull/1706/files#diff-bc13c6fff783e13d1577030cae7a3ae1c35d5e82b1fcab7ed639e3582bbf408cL23).

**Review by mikeminutillo — APPROVED** [review]

File: src/ServicePulse.Host/vue/src/components/DataView.vue (hunk 2 above)
Comment (johnsimons) [reply src/ServicePulse.Host/vue/src/components/DataView.vue:22]: const is fine @mikeminutillo

Comment (johnsimons) [reply src/ServicePulse.Host/vue/src/components/DataView.vue:22]: it defaults to `undefined`

File: src/ServicePulse.Host/vue/src/components/ExclamationMark.vue (hunk 3 above)
Comment (johnsimons) [reply src/ServicePulse.Host/vue/src/components/ExclamationMark.vue]: fixed

File: src/ServicePulse.Host/vue/src/components/DataView.vue (hunk 2 above)
Comment (mikeminutillo) [reply src/ServicePulse.Host/vue/src/components/DataView.vue:22]: I meant should it not be a `ref` at all. i.e.

```js
let refreshTimer : number;
```

Comment (johnsimons) [reply src/ServicePulse.Host/vue/src/components/DataView.vue]: fixed

File: src/ServicePulse.Host/vue/src/components/DataView.vue (hunk 1 above)
Comment (johnsimons) [reply src/ServicePulse.Host/vue/src/components/DataView.vue:11]: I am not 100% sure what you mean, @mikeminutillo .
Happy for you to refractor?

**Pull request merged**